test/integration/test_konnect_extension.go

```diff
@@ -39,40 +51,331 @@ func TestKonnectExtensionKonnectGatewayControlPlaneNamespacedRef(t *testing.T) {
 			authCfg.Spec.ServerURL = test.KonnectServerURL()
 		},
 	)
+	cert, key := certificate.MustGenerateSelfSignedCertPEMFormat()
 
-	cp := deploy.KonnectGatewayControlPlane(t, GetCtx(), clientNamespaced, authCfg,
-		deploy.WithTestIDLabel(testID),
-	)
+	t.Log("deploying backend deployment (httpbin) of HTTPRoute")
+	container := generators.NewContainer("httpbin", testutils.HTTPBinImage, 80)
+	deployment := generators.NewDeploymentForContainer(container)
+	require.NoError(t, clientNamespaced.Create(ctx, deployment))
+
+	t.Logf("exposing deployment %s via service", deployment.Name)
+	service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeClusterIP)
+	require.NoError(t, clientNamespaced.Create(ctx, service))
+
+	t.Run("Konnect hybrid ControlPlane", func(t *testing.T) {
+		// Create a Konnect control plane for the KonnectExtension to attach to.
+		cp := deploy.KonnectGatewayControlPlane(t, GetCtx(), clientNamespaced, authCfg,
+			deploy.WithTestIDLabel(testID),
+		)
+		t.Logf("Waiting for Konnect ID to be assigned to ControlPlane %s/%s", cp.Namespace, cp.Name)
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
+			err := GetClients().MgrClient.Get(GetCtx(), k8stypes.NamespacedName{Name: cp.Name, Namespace: cp.Namespace}, cp)
+			require.NoError(t, err)
+			assertKonnectEntityProgrammed(t, cp)
+		}, testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
+		// Order of deleting objects with finalizers:
+		// KongRoute & KongService -> DataPlane -> KonnectExtension -> Secret -> KonnectGatewayControlPlane.
+		// The first object deleted by calling `deleteObjectAndWaitForDeletionFn` will be deleted last when added by `CleanUp`,
+		// so the order of calling the deleting function should be a reverse of the order above.
+		// After they are all deleted, the namespace can be deleted in the final clean up.
+		t.Cleanup(deleteObjectAndWaitForDeletionFn(t, cp.DeepCopy()))
+
+		t.Log("Creating a KongService and a KongRoute to the service")
+		ks := deploy.KongService(t, ctx, clientNamespaced,
```

Comment:
```suggestion
		ks := deploy.KongService(t, ctx, clientNamespaced,
```

This also logs already: https://github.com/Kong/gateway-operator/blob/624554aa7ed45a31d973afa59d6321640153181d/test/helpers/deploy/deploy_resources.go#L357